### PR TITLE
Nb/feat/special chars check

### PR DESCRIPTION
**Pull request by Nadav-Barak**
Refactor of the Special Char check + plot file
[Special Characters — gallery-boilerplate documentation.pdf](https://github.com/deepchecks/deepchecks/files/11625780/Special.Characters.gallery-boilerplate.documentation.pdf)

Resolves https://linear.app/deepchecks/issue/DEE-624/change-thresholds-of-traintestsamplesmix-and-specialcharacters

### CODE REVIEW

File: deepchecks/nlp/checks/data_integrity/special_characters.py [hunk 1]
```diff
@@ -83,116 +89,81 @@ def run_logic(self, context: Context, dataset_kind) -> CheckResult:
...
-            message = ''
-        else:
-            message = (
-                f'Showing only the top {self.n_most_common} most common special characters, '
-                'you can change it using n_most_common param.'
-            )
+        display_table = pd.DataFrame(columns=['Sample ID', '% of Special Characters',
+                                              'Special Characters', 'Text'])
+        for idx, value in percent_special_chars_in_sample[:self.max_samples_to_show].items():
+            text_sample = dataset.get_sample_at_original_index(idx)
```
Comment (noamzbr) [inline deepchecks/nlp/checks/data_integrity/special_characters.py:126]: This is a bit wastful computationally. Maybe better option is to either do it all at once (guess numpy will be more efficient), or cache the location index seperatly

File: docs/source/checks/nlp/data_integrity/plot_special_characters.py [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+* `Define a Condition <#define-a-condition>`__
+
+Why check for special characters?
+===================================
+
+The ``SpecialCharacters`` check looks for text sample in which the percentage of special characters
+out of all characters is significant. Such samples can be an indicator for a problem in the data pipeline that
+require attention. Additionally, such examples may be problematic for the model to predict on.
+For example, a text sample with many emojis may be hard to
+predict on and a common mythology will be to replace them with a textual representation of the emoji.
```
Comment (noamzbr) [inline docs/source/checks/nlp/data_integrity/plot_special_characters.py:24]: Would replace with an example of html elements, for example, that may have made their way into our data.
If we're keeping the emoiji example, not sure this is the comon methodology to replace emojies with text. You have an example.
At the bare minumum - mythology -> methodolgy :P

File: docs/source/checks/nlp/data_integrity/plot_special_characters.py [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+# =============
+
+from deepchecks.nlp.checks import SpecialCharacters
+
+check = SpecialCharacters()
+result = check.run(text_data)
+result.show()
+
+#%%
+# In addition to the check display we can also see receive a summary of most common special characters
```
Comment (noamzbr) [inline docs/source/checks/nlp/data_integrity/plot_special_characters.py:48]: A word about what we're seeing here and in the result

File: docs/source/checks/nlp/data_integrity/plot_special_characters.py (hunk 2 above)
Comment (Nadav-Barak) [reply docs/source/checks/nlp/data_integrity/plot_special_characters.py:24]: This check is not important enough to add a new dataset example

File: deepchecks/nlp/checks/data_integrity/text_duplicates.py [hunk 4]
```diff
@@ -146,13 +146,12 @@ def run_logic(self, context: Context, dataset_kind):
...
         table = table.sort_values(by=['Number of Samples'], ascending=False)
-        table = table.set_index(['Sample IDs', 'Number of Samples'])
 
         return CheckResult(
             value=result_value,
             display=[
                 f'{format_percent(percent_of_duplicates)} of data samples are duplicates.',
                 'Each row in the table shows an example of a text duplicate and the number of times it appears.',
-                table
+                table.style.hide_index()
```
[the remark this answers is not in the document]
Comment (Nadav-Barak) [reply deepchecks/nlp/checks/data_integrity/text_duplicates.py:155]: works

File: deepchecks/nlp/checks/data_integrity/special_characters.py (hunk 1 above)
Comment (Nadav-Barak) [reply deepchecks/nlp/checks/data_integrity/special_characters.py:126]: It only run on the 3-10 sample that appear in the display

File: docs/source/checks/nlp/data_integrity/plot_special_characters.py (hunk 3 above)
Comment (Nadav-Barak) [reply docs/source/checks/nlp/data_integrity/plot_special_characters.py:48]: added

File: deepchecks/nlp/checks/data_integrity/text_duplicates.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (noamzbr) [reply deepchecks/nlp/checks/data_integrity/text_duplicates.py:155]: Matan says the problem here is with pandas 2.0.1, can you also test that? Sorry

**Review by noamzbr — APPROVED** [review]

File: deepchecks/utils/dataframes.py [hunk 5]
```diff
@@ -253,3 +254,13 @@ def cast_categorical_to_object_dtype(df: pd.DataFrame) -> pd.DataFrame:
     if categorical_columns:
         df = df.astype({c: 'object' for c in categorical_columns})
     return df
+
+
+def hide_index_for_display(df: t.Union[pd.DataFrame, pd.io.formats.style.Styler]) -> pd.io.formats.style.Styler:
+    """Hide the index of a dataframe for display."""
+    with warnings.catch_warnings():
```
Comment (matanper) [inline deepchecks/utils/dataframes.py:261]: remove this catch, we better know if there are warnings

**Pull request merged**